src/boot/lib/mlang.ml

```diff
@@ -718,14 +793,16 @@ let desugar_top (nss, langs, subs, syns, (stack : (tm -> tm) list)) = function
         (* TODO(vipa,?): this does not declare the type itself *)
         | Data (_, name, cdecls) ->
             List.fold_right (wrap_con name) cdecls tm
+        | Alias (fi, name, ty) ->
```

Comment:
So in my understanding, the `type`s are always emitted at the top, and this is not part of `wrap_data`. Because `aliases` are placed before `syns` in `decls` when the `Lang` is created on line 302 as
```
  Lang (info, name, includes, aliases @ List.rev_append syns inters)
```
I think this should guarantee that they are always placed before the `cons`, because `wrap_data` is called from a `fold_right`. This initialization is probably inefficient, but if I did `List.rev_append` for `aliases` I think they ended up in the wrong order relative to each other.